btcutil/psbt/taproot.go

```diff
@@ -113,9 +157,22 @@ func ReadTaprootBip32Derivation(xOnlyPubKey,
 		return nil, ErrInvalidPsbtFormat
 	}
 
-	// A hash is 32 bytes in size, so we need at least numHashes*32 + 5
-	// bytes to be present.
-	if len(value) < (int(numHashes)*32)+5 {
+	// As a safety/sanity check, verify that the hash count fits in a `uint32`.
+	// This isn’t mandated by BIP‑371, but it prevents overflow and limits
+	// derivations to about 137 GiB of data.
+	if numHashes > math.MaxUint32 {
+		return nil, ErrInvalidPsbtFormat
+	}
+
+	// Given the number of hashes, we can calculate the minimum byte size
+	// of the taproot BIP32 derivation.
+	minByteSize, err := minTaprootBip32DerivationByteSize(numHashes)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure that value is at least the minimum size.
+	if uint64(len(value)) < minByteSize {
```

Comment:
Okay, I guess this check prevents us from OOMing if the encoded length is larger from the number of bytes we actually have available.

Reply:
I think you reviewed before I managed to push my latest change. I've added:
```
	// This function allocates additional memory while parsing the serialized
	// data. To prevent potential out-of-memory (OOM) issues, we must validate
	// the length of the value slice before proceeding.
	if len(value) > MaxPsbtValueLength {
		return nil, ErrInvalidPsbtFormat
	}
```

let me know if that works for you